ztp/gitops-subscriptions/argocd/README.md

```diff
@@ -49,14 +49,68 @@ These steps configure your hub cluster with a set of ArgoCD Applications which g
 
 ### Deploying a site
 The following steps prepare the hub cluster for site deployment and initiate ZTP by pushing CRs to your GIT repository.
-- Add required secrets for site to the hub cluster. These resources must be in a namespace with a name matching the cluster name. In the example the cluster name & namespace is `test-sno`
-    1. Create secret for authenticating to the site BMC. Ensure the secret name matches the name used in the SiteConfig. In the example SiteConfig this is named `test-sno-bmh-secret`.
-    2. Create pull secret for site. The pull secret must contain all credentials necessary for installing OpenShift and all required operators. In the example SiteConfig this is named `assisted-deployment-pull-secret`
-- Add the SiteConfig CR for your site to your git repository
-    Note: The extra-manifest Machine configs exist under [source-crs/extra-manifest](https://github.com/openshift-kni/cnf-features-deploy/tree/master/ztp/source-crs/extra-manifest) will be included in the generated configMap for extra manifest.
-    Optional: For adding other extra-manifests to the provisioned cluster, create a directory ex; `sno-extra-manifest/` in relative path at siteconfig GIT repository and add the extra-manifest files to it. Then in the SiteConfig.yaml  set the extraManifestPath field ex; `extraManifestPath: sno-extra-manifest/` .
-- Add the PolicyGenTemplate CR for your site to your git repository
-- Push your changes to the git repository. The SiteConfig and PolicyGenTemplate CRs may be pushed simultaneously.
+1. Create the required secrets for site. These resources must be in a namespace with a name matching the cluster name. In out/argocd/example/siteconfig/example-sno.yaml the cluster name & namespace is `example-sno`
+   1. Create the namespace for the cluster:
+```
+$ export CLUSTERNS=example-sno
+$ oc create namespace $CLUSTERNS
+```
+   2. Create a pull secret for the cluster. The pull secret must contain all credentials necessary for installing OpenShift and all required operators. In all of the example SiteConfigs this is named `assisted-deployment-pull-secret`
+```
+$ oc apply -f - <<EOF
+apiVersion: v1
+kind: Secret
+metadata:
+  name: assisted-deployment-pull-secret
+  namespace: $CLUSTERNS
+type: kubernetes.io/dockerconfigjson
+data:
+  .dockerconfigjson: $(base64 <pull-secret.json)
+EOF
+```
+   3. Create a BMC authentication secret for each host you will be deploying.
+```
+$ oc apply -f - <<EOF
+apiVersion: v1
+kind: Secret
+metadata:
+  name: $(read -p 'Hostname: ' tmp; echo -n $tmp)-bmc-secret
+  namespace: $CLUSTERNS
+type: Opaque
+data:
+  username: $(read -p 'Username: ' tmp; base64 <<<$tmp)
+  password: $(read -s -p 'Password: ' tmp; base64 <<<$tmp)
```

Comment:
Do these commands ensure that the base64 encoded values does not have a trailing newline?

Reply:
> Do these commands ensure that the base64 encoded values does not have a trailing newline?

Good catch! The `<<<$tmp` appends a newline!  Will fix.